keras_hub/src/models/sam/sam_transformer.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2024 The KerasHub Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import keras
+from keras import ops
+
+from keras_hub.src.models.sam.sam_layers import (
+    MultiHeadAttentionWithDownsampling,
+)
+from keras_hub.src.models.sam.sam_layers import TwoWayMultiHeadAttention
+
+
+class TwoWayTransformer(keras.layers.Layer):
+    """A two-way cross-attention transformer decoder.
+
+    A transformer decoder that attends to an input image using
+    queries whose positional embedding is supplied.
+    The transformer decoder design is shown in [1]_. Each decoder layer
```

Comment:
Where is `[1]` that it's referred to here? Should there be a link instead of the `_` at the end?